src/metaswitch/clearwater/config_manager/config_access.py

```diff
@@ -765,6 +767,19 @@ def print_diff_and_syslog(config_type, config_1, config_2):
     Returns True if there are changes, that need to be uploaded, or False if
     the two are the same.
     """
+
+    # Check we've got a valid config type - this function might not be called
+    # directly from within the module.
+    config_classes = load_plugins_in_dir(PLUGIN_DIR)
```

Comment:
Hmmm.... I think doing it this way you are changing the functionality of `cw-restore_config`: Before your changes we were able to restore **every** config in etcd from **any** node type, However with this change I think we can only upload the **node-specific** config from the node that you're running `cw-restore_config` from.
At https://github.com/Metaswitch/clearwater-etcd/blob/dev/src/metaswitch/clearwater/config_manager/config_type_plugin_loader.py#L17 the function doc string says:
```
def load_plugins_in_dir(dir, params=None):
    """Loads plugins by:
        - looking for all .py files in the given directory
        - calling their load_as_plugin() function
        - returning a list containing the return values of all load_as_plugin()
        calls
        """
```
So it's literally just looking for files present in `/usr/share/clearwater/clearwater-config-access/plugins/` which is different on different node types, e.g. a DGN only has two files:
```
[dgn]clearwater@oamp-auditd-site1-dgn-1:~$ ls -l /usr/share/clearwater/clearwater-config-access/plugins/
total 8
-rw-r--r-- 1 root root 1170 Nov  6 11:14 dns_json_config_plugin.py
-rw-r--r-- 1 root root 1108 Nov  6 11:14 shared_config_config_plugin.py
```

I don't think that this is a good use case: I could imagine customers having regular backups and if shit really hits the fan they want to have their whole deployment in a good state again without figuring out what exact config they need to apply again. It could also happen that all instances of a particular node type have died so then you wouldn't have any chance to restore that config in etcd.